java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -143,22 +144,24 @@ private void save(UnitTestIndex index, SensorContext context) {
     }
   }
 
-  private void save(UnitTestClassReport report, Resource resource, SensorContext context) {
-    double testsCount = report.getTests() - report.getSkipped();
-    saveMeasure(context, resource, CoreMetrics.SKIPPED_TESTS, report.getSkipped());
-    saveMeasure(context, resource, CoreMetrics.TESTS, testsCount);
-    saveMeasure(context, resource, CoreMetrics.TEST_ERRORS, report.getErrors());
-    saveMeasure(context, resource, CoreMetrics.TEST_FAILURES, report.getFailures());
-    saveMeasure(context, resource, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
+  private void save(UnitTestClassReport report, InputFile inputFile, SensorContext context) {
+    int testsCount = report.getTests() - report.getSkipped();
+    saveMeasure(context, inputFile, CoreMetrics.SKIPPED_TESTS, report.getSkipped());
+    saveMeasure(context, inputFile, CoreMetrics.TESTS, testsCount);
+    saveMeasure(context, inputFile, CoreMetrics.TEST_ERRORS, report.getErrors());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_FAILURES, report.getFailures());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
     double passedTests = testsCount - report.getErrors() - report.getFailures();
     if (testsCount > 0) {
-      double percentage = passedTests * 100d / testsCount;
-      saveMeasure(context, resource, CoreMetrics.TEST_SUCCESS_DENSITY, ParsingUtils.scaleValue(percentage));
+      double percentage = ParsingUtils.scaleValue(passedTests * 100d / testsCount);
+      if (!Double.isNaN(percentage)) {
+        context.<Double>newMeasure().forMetric(CoreMetrics.TEST_SUCCESS_DENSITY).on(inputFile).withValue(percentage).save();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)